### ISSUE TITLE
To be able to query last lines of logs in Loki 

### ISSUE THREAD

**Issue by isaiasanchez** [grafana/loki#6315]
**Is your feature request related to a problem? Please describe.**
We need to query the last n lines of scrapped logs not just the last n hours, because if a problem occur the logs stop been generated and checking the last hour for example you see an error, but next hour you'll see all ok because there are no more log lines. 

**Describe the solution you'd like**
What I want if to be able to check last n lines of the log even if it's not been generated anymore.

**Describe alternatives you've considered**
The only alternative I've considered is to check more time that necessary, for example if the check is every last hour, change it to check last 12 hours in that way we'll receive the error more times, but it's not a good solution because we'll receive alarms even if the problem is already solved.

**Additional context**

**Comment by siredmar** [comment]
@isaiasanchez did you find any solution for this? I'm facing the same issue.

**Comment by isaiasanchez** [comment]
no, i didn't. For now we just check the last n hours of logs looking for a match, then if the log file isn't continuous these match disappear.  It isn't ideal in our case.

Seems this request is not something Grafana Labs will consider to include.

**Comment by quantumsheep** [comment]
Hey 👋 I have the same need, is there any news on this?

**Comment by hide-in-code** [comment]
+1

**Comment by Spinestars** [comment]
+1
